src/setup/node.rs

```diff
@@ -74,13 +77,36 @@ impl Node {
         self
     }
 
+    /// Sets whether to signal the node has started through a peer connection.
+    ///
+    /// If set to true, the call to [`start`] will initiate a listener set as a peer on the node and
+    /// will only return once it has received a connection request. This isn't necessary in
+    /// scenarios in which the node initates the connections.
+    ///
+    /// [`start`]: methode@Node::start
+    pub fn signal_when_started(&mut self, addr: SocketAddr) -> &mut Self {
```

Comment:
I'm trying to think of a name that better describes that we are waiting for an initial connection signal before returning.

`block_until_connected` ? `wait_until_connected` ?